pos_meal_voucher/static/src/js/models.js

```diff
@@ -39,6 +39,52 @@ odoo.define("pos_meal_voucher.models", function (require) {
 
     models.Order = Order;
 
+    var Orderline = models.Orderline.extend({
+        // OVERWRITE the function generate_wrapped_product_name
+        // Because there is no way to overload it correctly.
+
+        generate_wrapped_product_name: function() {
+            var MAX_LENGTH = 24; // 40 * line ratio of .6
+            var wrapped = [];
+            var name = this.get_product().display_name;
+
+            // <Begin of the patch>
+            if (this.get_product().meal_voucher_ok) {
+                name = "(*) " + name;
+            }
+            // <End of the patch>
+
+            var current_line = "";
+
+            while (name.length > 0) {
+                var space_index = name.indexOf(" ");
+
+                if (space_index === -1) {
+                    space_index = name.length;
+                }
+
+                if (current_line.length + space_index > MAX_LENGTH) {
+                    if (current_line.length) {
+                        wrapped.push(current_line);
+                    }
+                    current_line = "";
+                }
+
+                current_line += name.slice(0, space_index + 1);
+                name = name.slice(space_index + 1);
+            }
+
+            if (current_line.length) {
+                wrapped.push(current_line);
+            }
+
+            return wrapped;
```

Comment:
```suggestion

            if (!this.get_product().meal_voucher_ok) { // TODO: check for config, too?
                return OrderlineSuper.generate_wrapped_product_name.apply(this, arguments);
            }
            const originalDisplayName = this.product.display_name;
            this.product.display_name = "(*) " + originalDisplayName;
            const res = OrderlineSuper.generate_wrapped_product_name.apply(this, arguments);
            this.product.display_name = originalDisplayName;
            return res;
```

There is this hackish alternative that usually just works.
Kind of temporarily monkey-patching to achieve what you want, but always making sure to revert to the original at the end.
Works fine in synchronous functions, never to be used on async functions though. ⚠️